### PR TITLE
Autodetect some transform properties.

### PR DESCRIPTION
**Pull request by anntzer**
- 1d transforms are always separable, no need to repeat `is_separable =
  True` again and again.
- If a subclass overrides `inverted()` then it is most likely invertible
  (though not always), no need to repeat `has_inverse = True` again and
  again (but subclasses can always override to `has_inverse = False`).

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by tacaswell** [comment]
How often do we check these in the code, iirc the the performance cost of properties over attribute access is sizable.

**Comment by tacaswell** [comment]
This actually seems like a good case for `__init_subclass__`....

**Comment by anntzer** [comment]
Good point re `__init_subclass__`, now implemented, which should also handle any worries about performance cost.

File: lib/matplotlib/transforms.py [hunk 1]
```diff
@@ -1191,9 +1191,9 @@ class Transform(TransformNode):
     - :attr:`input_dims`
     - :attr:`output_dims`
     - :meth:`transform`
-    - :attr:`is_separable`
-    - :attr:`has_inverse`
-    - :meth:`inverted` (if :attr:`has_inverse` is True)
+    - :attr:`is_separable` (defaults to True for 1d -> 1d transforms)
```
Comment (timhoffm) [inline lib/matplotlib/transforms.py]: > Subclasses of this class should override the following members (at minimum):

Is not correct anymore for `is_separable` and `has_inverse`. Either rephrase or put those two in a separate block.

Comment (anntzer) [reply lib/matplotlib/transforms.py]: good point, done

**Pull request merged**